### ISSUE TITLE
Rancher on Docker NodePort does not work

### ISSUE THREAD

**Issue by josefjahn** [rancher/rancher#32039]
**What kind of request is this (question/bug/enhancement/feature request):** question / bug

**Steps to reproduce (least amount of steps as possible):**
Run rancher on a single linux node using docker.
Follow the rancher docs at https://rancher.com/docs/rancher/v2.5/en/quick-start-guide/workload/quickstart-deploy-workload-nodeport/ and deploy a simple workload in the "local" cluster.
Expose that workload using nodeport.

**Result:**
Workload cannot be accessed. Nodeport is configured in cluster explorer, but doesn't work and is also not listened to in the host OS. 
It should be noted that I only have this issue on k3s clusters. If I expose a port in an existing cluster created with pre-2.5 Rancher, there are no problems.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.7
- Installation option (single install/HA): single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): metal, 4GB RAM
- Kubernetes version (use `kubectl version`): 

```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8+k3s1", GitCommit:"6b595318666804506c19cfc3d3d228423d38fab1", GitTreeState:"clean", BuildDate:"2020-08-14T20:42:29Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.13.8
 Git commit:        afacb8b7f0
 Built:             Fri Dec 18 12:15:19 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       afacb8b7f0
  Built:            Fri Dec  4 23:02:49 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.3-0ubuntu2.3
  GitCommit:
 runc:
  Version:          spec: 1.0.1-dev
  GitCommit:
 docker-init:
  Version:          0.18.0
  GitCommit:

```

![image](https://user-images.githubusercontent.com/23048489/114285638-05806700-9a59-11eb-9e52-30062795f6ab.png)

![image](https://user-images.githubusercontent.com/23048489/114285647-14ffb000-9a59-11eb-828c-467094c056b8.png)

![image](https://user-images.githubusercontent.com/23048489/114285671-352f6f00-9a59-11eb-8ab3-94732281e843.png)

**Comment by josefjahn** [comment]
To clarify, I get the same issue if I create a new cluster (instead of using local). Also, clicking on the nodeport link takes me to 172.17.0.2:xxxxx instead of my host machine's local IP.

**Comment by vincent99** [comment]
When you run Rancher in docker is running a k3s cluster inside the container.  This is intended only to run Rancher, and the ports exposed don't work because you haven't also exposed them through docker.

**Comment by josefjahn** [comment]
Understood, but if I create a new cluster, add my node to it, and install a workload there I would expect nodeport to work from the outside, like it did pre-2.5
I feel like I am missing a step here which is new to 2.5.

I have multiple clusters and upgrading them worked without a problem. But if I do a fresh install, I cannot reach my workloads.

**Comment by vincent99** [comment]
This is almost always some sort of an "oops, firewall", or security group, or the overlay network doesn't work, etc situation

I'm sorry but 2.5.x has been out for 6 months, with thousands and thousands of clusters and installs, the problem is not "NodePorts don't work in 2.5".  The version of Rancher really has little to do with the operation of k8s features in a running cluster..

**Comment by josefjahn** [comment]
I can't reproduce the problem anymore :/

I've performed a test on the same box using Rancher v2.4.2:
1) docker run -d -p 80:80 -p 443:443 rancher/rancher:v2.4.2
2) Created cluster "test" (which uses Kubernetes v1.17.5)
3) Copy/pasted the docker command to join the node into the cluster with all roles
4) Deployed the hello world workload with nodeport on 30001

Now I tried the same with v2.5.2:

1) Wipe the previous installation using the procedure from documentation at https://rancher.com/docs/rancher/v2.x/en/cluster-admin/cleaning-cluster-nodes/
2) docker run -d -p 80:80 -p 443:443 --privileged rancher/rancher:latest
3) Created cluster "test25" (Kubernetes v1.20.5)
4) Copy/paste the docker command to add the node to the cluster

Result: Works as with 2.4.2.


Finally I thought, what did I do differently last time? I first tried to add a workload to the local project, which as you pointed out would require me to expose the ports in docker, and THEN I created a new cluster and added the workload there. Infuriatingly that worked as well. Even the nodeport link in the UI was correct (192.168.178.11 instead of the 172 cluster ip)

Bottom line: I cannot reproduce the problem. I know for sure it was there, as was the incorrect link to the IP:nodeport of the worload, and I am certain I was working in the test cluster I created, not local. But since I can't reproduce it, I'm closing this ticket.

Thanks for the quick response, sorry for the trouble!
